scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -173,6 +174,15 @@ object TreeOps {
         case t: Ctor.Secondary =>
           addDefn[KwDef](t.mods, t)
           addAll(t.stats)
+        // special handling for rewritten blocks
+        case t @ Term.Block(List(_)) // single-stat block
```

Comment:
Why are these two cases needed? I am having difficulty following the logic here for `replacedWith` and what situation this is influenced by.

Reply:
@tgodzik my apologies, i should have explained.

the logic for optional braces rewrite is different from the one we employed with older rewrites. the old rewrites traversed the tree, then collected and applied some token patches, converted to a string and re-parsed the code again.

the new approach is different. we parse the tree, create a `FormatTokens` object (a sequence of `FormatToken` pairs, with intervening whitespace) and then modify this sequence ONLY (no changes in trees or parsing).

therefore, this particular logic, which determines the start of a statement (but in reality is used for two primary purposes: to find where one statement ENDS and another BEGINS, and to find the beginning of a lambda enclosed in braces), breaks down a little bit.

for instance, if we remove an optional brace from from the start of a block, the `Router` rule which would have formatted `{?` before no longer applies; unless we ignore this rewritten block here, the wrong rule (which looks at starts of statements and was never intended to look block starts) would apply.

this is the minimum change to make this work. the method `extractStatementsIfAny` (line 80 above) is also used for a completely different purpose, or else we could have changed it to return `block.stats.tail` or `match.cases.tail` (although that still doesn't solve the problem with brace-enclosed lambdas).